controllers/managedocs_controller.go

```diff
@@ -177,3 +170,34 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) generateStorageClusterWithParams(configSecret *corev1.Secret) (*ocsv1.StorageCluster, error) {
+	sdata := configSecret.Data
+	requestedStorageCapacity, err := resource.ParseQuantity(string(sdata[storageClassSizeKey]))
+	if err != nil {
+		r.Log.Error(err, "Missing or incorrect parameter in the addonparam secret", "Key", storageClassSizeKey)
+		return nil, err
+	}
+	r.Log.Info("Addon param data", "Storage Cluster Size", requestedStorageCapacity)
+
+	osdSize, _ := resource.ParseQuantity("1Ti")
+
+	// The request must be an exact multiple of the base OSD size
+	rem := requestedStorageCapacity.Value() % osdSize.Value()
+
+	if rem != 0 {
+		err := fmt.Errorf("Storage cluster size must be a multiple of %v", osdSize)
+		return nil, err
+	}
+	count := requestedStorageCapacity.Value() / osdSize.Value()
```

Comment:
Why aren't we just treating the size param as a number which is a multiplication of 1Tib. If we do we will have no fail conditions

Reply:
Treating the input as an integer will still fail if a non-integer is passed in. That check will need to be enforced in the UI